web/src/test/java/org/springframework/security/web/firewall/StrictHttpFirewallTests.java

```diff
@@ -41,6 +44,19 @@ public class StrictHttpFirewallTests {
 
 	private MockHttpServletRequest request = new MockHttpServletRequest("GET", "");
 
+	private static final Pattern ASSIGNED_AND_NOT_ISO_CONTROL_PATTERN = Pattern
+		.compile("[\\p{IsAssigned}&&[^\\p{IsControl}]]*");
+
+	private static final Pattern HEADER_VALUE_PATTERN = Pattern.compile("[\\p{IsAssigned}&&[[^\\p{IsControl}]||\\t]]*");
+
+	@BeforeEach
+	public void initializeFirewall() {
+		StrictHttpFirewall.ALLOWED_HEADER_NAMES = (s) -> ASSIGNED_AND_NOT_ISO_CONTROL_PATTERN.matcher(s).matches();
```

Comment:
I think these will already be set by the object itself. They won't need to be reset if you follow the earlier suggestions.

Reply:
Done.